BLSQ/iaso
# IA-3093: Bug with roles

Explain what problem this PR is resolving
- Bug with roles

Related JIRA tickets : [IA-3093](https://bluesquare.atlassian.net/browse/IA-3093)

## Self proofreading checklist


## Changes
- Remove cache on dropdown user roles
- Remove all users in the group and delete the group before deleting the user role
- Add new user role and check if it trigger a dialog for warning the user

## How to test

- Create a user role and check if it's available directly in dropdown
- Delete a user role and after try to create another one with the same name, before it wouldn't work

## Print screen / video
[Screencast from 2024-07-08 12-31-39.webm](https://github.com/BLSQ/iaso/assets/19631540/5caeaa6b-54bc-4074-907a-9ce34f7ef177)



[IA-3093]: https://bluesquare.atlassian.net/browse/IA-3093?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
